src/gui/qgsbrowserdockwidget.cpp

```diff
@@ -197,7 +197,18 @@ void QgsBrowserDockWidget::showContextMenu( QPoint pt )
     menu->addAction( tr( "Add a Directory..." ), this, SLOT( addFavoriteDirectory() ) );
   }
 
+  const QList<QMenu *> menus = item->menus();
```

Comment:
I think we have two possibilities here:

* Transfer ownership to the caller and set the parent right at this place here
* Provide the `menus()` method with a proper parent by calling it with `item->menus( menu )`

Reply:
ok, I will go with this, thanks!